src/svgdevicecontext.cpp

```diff
@@ -163,6 +163,11 @@ void SvgDeviceContext::StartGraphic(Object *object, std::string gClass, std::str
         }
     }
 
+    // make sure we are in a g container
+    while (strcmp(m_currentNode.name(), "g")) {
+        m_currentNode = m_currentNode.parent();
+    }
```

Comment:
This assures correct SVG structure in general and is needed for annotations within text elements. It just pushes the `g.annot` up to the parent `g`. (Tested thoroughly!)